examples/wavenet.cc

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include <ATen/ATen.h>
+
+#include "tc/aten/aten_compiler.h"
+#include "tc/autotuner/genetic_autotuner_aten.h"
+#include "tc/core/cuda/cuda_mapping_options.h"
+#include "tc/core/flags.h"
+
+#include "../test/test_harness_aten_cuda.h"
+
+DEFINE_string(tuner_proto, "", "Filename to load and store proto cache ");
+
+TEST(WaveNet2Layers, SimpleAutotune) {
+  // 1. Define and setup the TC compilation unit with CUDA memory
+  // management backed by ATen tensors.
+  std::string tc = R"TC(
+def wavenet2layers(float(OUT, IN, KERN) Weight0,
+    float(OUT) Bias0,
+    float(BATCH, IN, KERN) Data0,
+    float(IN, IN) ResWeight0,
+    float(IN) ResBias0,
+    float(SKIP, IN) SkipWeight0,
+    float(SKIP) SkipBias0,
+    float(OUT, IN, KERN) Weight1,
+    float(OUT) Bias1,
+    float(BATCH, IN, KERN) Data1,
+    float(IN, IN) ResWeight1,
+    float(IN) ResBias1,
+    float(SKIP, IN) SkipWeight1,
+    float(SKIP) SkipBias1)
+    -> (Res0, Dilate0, NonLin0, Skip0, Res1, Dilate1, NonLin1, Skip1)
+{
+    Dilate0(batch, out)   =   Bias0(out)
+            where batch in 0:BATCH
+    Dilate0(batch, out)  += Weight0(out, r_in, r_kern) * Data0(batch, r_in, r_kern)
+    NonLin0(batch, out)   = 1 / (1 + exp(-1*(Dilate0(batch, out))))
+    NonLin0(batch, out)  *= tanh(Dilate0(batch, out + 64))
+      Skip0(batch, skip)  =   SkipBias0(skip)
```

Comment:
The parentheses on the RHS are not aligned with the previous statement, but are aligned with the following one, so I'd add an empty line before this expression.